tests/migrations/test_autodetector.py

```diff
@@ -2181,6 +2181,150 @@ def test_rename_model_reverse_relation_dependencies(self):
+        before = [self.author_name, self.book]
+        after = [
+            ModelState(
+                "otherapp",
+                "Author",
+                [
+                    ("id", models.AutoField(primary_key=True)),
+                    ("name", models.CharField(max_length=200)),
+                ],
+            ),
+            ModelState(
+                "otherapp",
+                "Book",
+                [
+                    ("id", models.AutoField(primary_key=True)),
+                    ("author", models.ForeignKey("otherapp.Author", models.CASCADE)),
+                    ("title", models.CharField(max_length=200)),
+                ],
+            ),
+        ]
+        changes = self.get_changes(
+            before, after, MigrationQuestioner({"ask_move_model": True})
+        )
+        self.assertNumberMigrations(changes, "otherapp", 2)
+
+        self.assertMigrationDependencies(changes, "otherapp", 0, [])
+        self.assertOperationTypes(changes, "otherapp", 0, ["CreateModel"])
+        self.assertMigrationDependencies(
+            changes, "otherapp", 1, [("otherapp", "auto_1"), ("testapp", "auto_1")]
+        )
+        self.assertOperationTypes(
+            changes, "otherapp", 1, ["AlterModelOptions", "AlterField"]
+        )
+
+        self.assertNumberMigrations(changes, "testapp", 2)
+
+        self.assertMigrationDependencies(
+            changes, "testapp", 0, [("otherapp", "auto_1")]
+        )
+        self.assertOperationTypes(
+            changes, "testapp", 0, ["AlterModelTable", "AlterModelOptions"]
+        )
+        self.assertMigrationDependencies(
+            changes, "testapp", 1, [("otherapp", "auto_2"), ("testapp", "auto_1")]
+        )
+        self.assertOperationTypes(changes, "testapp", 1, ["DeleteModel"])
+
+    def test_move_model_with_db_table_changed(self):
+        before = [self.author_with_db_table_options, self.book]
+        after = [
+            ModelState(
+                "otherapp",
+                "Author",
+                [
+                    ("id", models.AutoField(primary_key=True)),
+                    ("name", models.CharField(max_length=200)),
+                ],
+                {"db_table": "changed_db_table"},
+            ),
+            ModelState(
+                "otherapp",
+                "Book",
+                [
+                    ("id", models.AutoField(primary_key=True)),
+                    ("author", models.ForeignKey("otherapp.Author", models.CASCADE)),
+                    ("title", models.CharField(max_length=200)),
+                ],
+            ),
+        ]
+        changes = self.get_changes(
+            before, after, MigrationQuestioner({"ask_move_model": True})
+        )
+        self.assertNumberMigrations(changes, "otherapp", 2)
+
+        self.assertMigrationDependencies(changes, "otherapp", 0, [])
+        self.assertOperationTypes(changes, "otherapp", 0, ["CreateModel"])
```

Comment:
It'd be good to test that the `CreateModel` has a `db_table: "changed_db_table"`